src/System.CommandLine.Tests/EnvironmentVariableDirectiveTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.CommandLine.Tests
+{
+    public class EnvironmentVariableDirectiveTests
+    {
+        private static readonly Random randomizer = new Random(Seed: 456476756);
+
+        [Fact]
+        public static async Task Sets_environment_variable_to_value()
+        {
+            bool asserted = false;
+            string variable = $"TEST_ENVIRONMENT_VARIABLE{randomizer.Next()}";
```

Comment:
Rather than repeating this setup, you could assign the environment variable name to a field in the constructor.

Reply:
51bd07b